fenix/app/src/main/java/org/mozilla/fenix/compose/SwitchWithLabel.kt

```diff
@@ -47,9 +50,16 @@ fun SwitchWithLabel(
     enabled: Boolean = true,
 ) {
     Row(
-        modifier = modifier,
         horizontalArrangement = Arrangement.spacedBy(16.dp),
         verticalAlignment = Alignment.CenterVertically,
+        modifier = modifier.then(
```

Comment:
If it's only for talkback to read it as one component and not have the whole Row/component toggleable, we should use `.semantics(mergeDescendants = true) { }` modifier.

Reply:
I think all the element should be toggeable just like in Settings Screen.

Reply:
This request comes from this pr https://github.com/mozilla-mobile/firefox-android/pull/4081#discussion_r1364976300

Reply:
Thanks for the context, since this is an existing ui component that's ported to compose I'm assuming UX is on the same page.

```suggestion
        modifier = modifier
            .toggleable(
                value = checked,
                enabled = enabled,
                role = Role.Switch,
                onValueChange = onCheckedChange,
            )
            .semantics(mergeDescendants = true) {},
```
1. Here we don't need to use .then but can simply add the toggleable modifier. 
2. Let's add the enabled param since we want the whole time to be disabled based on the passed in param. 
3. `onValueChange = onCheckedChange` same pattern in the file.
4. I think `mergeDescedants` would be better than clearing semantics of the switch. If we add another composable in the row, this would handle it.

Reply:
I made the changes without  .semantics(mergeDescendants = true) {},